src/form.rs

```diff
@@ -76,9 +76,12 @@ impl Form for MetaForm {
 ///
 /// `type String` is owned in order to enable decoding
 #[derive(PartialEq, Eq, PartialOrd, Ord, Clone, Copy, Serialize, Debug)]
-pub enum CompactForm {}
+pub struct CompactForm<S = &'static str>(PhantomData<S>);
```

Comment:
For my education, can you elaborate on the choice of default here? Why is `&'static str` the natural choice here?

Reply:
When constructing type infos the strings are always `'static`. Only for deserializing/decoding by consumers do we need owned strings.